Source/qol/stash.cpp

```diff
@@ -248,7 +248,7 @@ void StartGoldWithdraw()
 {
 	CloseGoldDrop();
 
-	InitialWithdrawGoldValue = Stash.gold;
+	InitialWithdrawGoldValue = RoomForGold();

```

Comment:
```suggestion
	InitialWithdrawGoldValue = std::min(RoomForGold(), Stash.gold);
```
or just `Stash.gold` depending on the decision from the other comment trail :D